Json.More/JsonSerializerOptionsExtensions.cs

```diff
@@ -51,9 +52,204 @@ public static JsonConverter<T> GetConverter<T>(this JsonSerializerOptions option
 	/// <param name="value">The value to serialize.</param>
 	/// <param name="typeInfo">An explicit typeInfo to use for looking up the Converter. If not provided, options.GetTypeInfo will be used.</param>
 	/// <returns>The value that was converted.</returns>
-	public static void Write<T>(this JsonSerializerOptions options, Utf8JsonWriter writer, T? value, JsonTypeInfo<T>? typeInfo)
+	public static void Write<T>(this JsonSerializerOptions options, Utf8JsonWriter writer, T value, JsonTypeInfo<T> typeInfo)
```

Comment:
Not sure if I mentioned this in another PR, but the `options` and `typeInfo` parameters defined here are redundant and mixing them together can result in consistency problems. Either resolve the `typeInfo` from the `options.GetTypeInfo` method or get the `Options` instance from the `typeInfo` parameter.

Reply:
The TypeInfo here is used just to get the Converter, and we're using a JsonTypeInfo as parameter to make sure the caller has tagged the type on a JsonSerializerContext. The Options dictionary is used for all caller-defined types and any other configurations.

Reply:
(I'm still first on why the your info is tied to an options in the first place. They seem to be independent concepts.)